src/bin/config-store.rs

```diff
@@ -101,11 +101,11 @@ fn main() {
             }
         }
         Commands::Set { key, value } => {
-            store.set(&key, Setting::from_str(&value).expect("incorrect value"));
+            store.set(&key, Setting::from_string(&value).expect("incorrect value"));
```

Comment:
Curious why this was reverted.  The name `from_str` is common in Rust, while the name `from_string` looks like someone didn't know about `from_str`.

Reply:
@emwalker this is due to an warning that is emitted by rust itself... it clashes with a standard `from_str` that is found in a ::std trait and could cause conflicts

Reply:
Ah, gotcha.  Perhaps that means we should be implementing that trait instead of defining `from_string`.  It would not be hard.

```rust
impl FromStr for Setting {
  fn from_str(value: &str) -> Self {
    todo!()
  }
}
```